File: Go/Dmitriy-Vas/Fizz Buzz/fizzbuzz.go
package Fizz_Buzz

import "strconv"

func FizzBuzz(i int) string {
	out := ""
	if i % 3 == 0 {
		out += "Fizz"
	}
	if i % 5 == 0 {
		out += "Buzz"
	}
	if out == "" {
		out = strconv.Itoa(i)
	}
	return out
}

File: Haskell/Dmitriy-Vas/Fizz Buzz/fizzbuzz.hs
module Fizz_Buzz where

fizzBuzz :: Int -> String
fizzBuzz a | mod15 = "FizzBuzz"
           | mod3 = "Fizz"
           | mod5 = "Buzz"
           | otherwise = show a
  where mod3 = mod a 3 == 0
        mod5 = mod a 5 == 0
        mod15 = mod a 15 == 0
